test/e2e/framework/test_context.go

```diff
@@ -561,4 +554,39 @@ func AfterReadingAllFlags(t *TestContextType) {
 		}
 		os.Exit(1)
 	}
+
+	if TestContext.ReportDir != "" {
+		ginkgo.ReportAfterSuite("Kubernetes e2e JUnit report", writeJUnitReport)
+	}
+}
+
+// writeJUnitReport generates a JUnit file in the e2e report directory that is
+// shorter than the one normally written by `ginkgo --junit-report`. This is
+// needed because the full report can become too large for tools like Spyglass
+// (https://github.com/kubernetes/kubernetes/issues/111510).
+//
+// Users who want the full report can use `--junit-report`.
+func writeJUnitReport(report ginkgo.Report) {
+	trimmedReport := report
+	trimmedReport.SpecReports = nil
+	for _, specReport := range report.SpecReports {
+		// Remove details for any spec that hasn't failed. In Prow,
+		// the test output captured in build-log.txt has all of this
+		// information, so we don't need it in the XML.
+		if specReport.State != types.SpecStateFailed {
+			specReport.CapturedGinkgoWriterOutput = ""
+			specReport.CapturedStdOutErr = ""
+			// TODO: this is a bit too broad (removes *all* ReportEntries,
+			// not just the useless ones from ginkgo.By).
```

Comment:
I kept the code which trims just the ginkgo.By entries out of this PR to make it equivalent to https://github.com/kubernetes/kubernetes/pull/111627, but IMHO it would be better to fix this TODO right away.

Reply:
we are writing a new reporter, I think that is better if we merge the new reporter in one shot , if we are going to trim ginkgo.By it makes sense to do it now.

That also helps in case we have to roll back, better revert 1 commit than 2 , or one that depends on other

Reply:
I agree. I've force-pushed the additional change.

Reply:
I will link to this one to the issue: https://github.com/kubernetes/kubernetes/issues/109744